chain/network/src/peer_manager/tests/fuzzers.rs

```diff
@@ -0,0 +1,50 @@
+use crate::peer_manager::peer_manager_actor::Event as PME;
+use crate::peer_manager::testonly::Event;
+use crate::{network_protocol, peer_manager, testonly::make_rng};
+use near_async::time::Duration;
+use near_async::time::FakeClock;
+use std::future::Future;
+use std::sync::Arc;
+use tokio::io::AsyncWriteExt;
+
+fn in_tokio(f: impl Future<Output = ()>) {
+    tokio::runtime::Runtime::new().unwrap().block_on(f)
+}
+
+async fn random_handshake_connect(input: &[u8]) {
```

Comment:
Would it be valuable to include in tests a call to this with a valid handshake to check that it works as expected? We could return a status instead of panicking and then handle the return value appropriately outside the call.

Reply:
Definitely! Do you already know of binary encodings of a valid / invalid handshake, and/or where should I look for the detailed encoding of handshakes?

Also, I’m definitely not planning on leaving this panic, it’s just here to validate that the fuzzer can actually reach a valid handshake and is not blocked by eg. signature verification — based on the fuzzing on my dev machine during my time off as well as on the 24/7 fuzzer infra up to now, it looks like this is not the case and I’ll have some more work to do around there already.